src/Eshopworld.WorkerProcess/LeaseAllocator.cs

```diff
@@ -98,7 +105,7 @@ private async Task<LeaseStoreResult> TryPersistLease(ILease lease)
             // Read the lease for event
             lease = await _leaseStore.ReadByLeaseTypeAsync(_options.Value.WorkerType).ConfigureAwait(false);
 
-            _telemetry.Publish(new LeaseAcquisitionEvent(lease.InstanceId.Value, _options.Value.WorkerType,
+            _telemetry.Publish(new LeaseAcquisitionEvent(instanceId, _options.Value.WorkerType,
```

Comment:
Later on we will be adding check if the instanceid is matching and in such case, we are also good to go  (CC @nikhilgirraj )